app/src/main/java/com/manage/ak/moneyreport/MainActivity.java

```diff
@@ -0,0 +1,441 @@
+package com.manage.ak.moneyreport;
+
+import android.app.Dialog;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private Context context = this;
+
+    // A list to store bank list and cash sms
+    private List<Sms> bankList = new ArrayList<>();
+    private List<Sms> cashList = new ArrayList<>();
+
+    // Main balance in the bank
+    // In many places the balance is needed in String form.
+    private String BALANCE = "0.0";
+    // Total cash spent from cash in hand
+    private String CASHSPENT = "0.0";
+
+    // variables used to store and calculate balance for the cash transactions
+    private String amt, type, balance;
+
+    // bank balance and date TextView inside the bank card
+    private TextView bankBalance;
+    private TextView estimateDate;
+
+    // spent amount and cash in hand inside the cash card
+    private TextView spentAmount;
+    private TextView cashBalance;
+
+    // BroadcastReceiver listening to the incoming messages
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    IntentFilter intentFilter = new IntentFilter(SMS_RECEIVED);
+    private BroadcastReceiver sms_receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction().equals(SMS_RECEIVED)) {
+                readMessages();
```

Comment:
Its not reading the whole inbox. I have set a filter inside the
readMessages() method.

On Wed, Oct 28, 2015 at 12:33 PM, Shubham Goyal <notifications@github.com>
wrote:

> In app/src/main/java/com/manage/ak/moneyreport/MainActivity.java
> <https://github.com/ArtooTrills/Bounty-Android-Personal-Finance/pull/4#discussion_r43223003>
> :
>
> > +    // bank balance and date TextView inside the bank card
> > +    private TextView bankBalance;
> > +    private TextView estimateDate;
> > +
> > +    // spent amount and cash in hand inside the cash card
> > +    private TextView spentAmount;
> > +    private TextView cashBalance;
> > +
> > +    // BroadcastReceiver listening to the incoming messages
> > +    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
> > +    IntentFilter intentFilter = new IntentFilter(SMS_RECEIVED);
> > +    private BroadcastReceiver sms_receiver = new BroadcastReceiver() {
> > +        @Override
> > +        public void onReceive(Context context, Intent intent) {
> > +            if (intent.getAction().equals(SMS_RECEIVED)) {
> > +                readMessages();
>
> Why are you reading the complete inbox again on just receiving a single
> message ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ArtooTrills/Bounty-Android-Personal-Finance/pull/4/files#r43223003>
> .
>